common/src/main/java/dev/xhyrom/lighteco/common/model/user/User.java

```diff
@@ -67,6 +73,11 @@ public void setBalance(@NonNull Currency currency, @NonNull BigDecimal balance,
 
         if (!force)
             this.setDirty(true);
+
+        if (publish) {
```

Comment:
send only if publish is true and if user is not online

Reply:
update ConnectionListener PlayerQuitEvent - don't unload, just save user and after it send message via messenger

^ this will fix possible race condition